src/main/scala/com/ironcorelabs/document/DocumentId.scala

```diff
@@ -0,0 +1,13 @@
+package com.ironcorelabs.scala.sdk
+
+import cats.effect.Sync
+import com.ironcorelabs.{sdk => jsdk}
+
+/** ID of a document. Unique within the segment. Must match the regex `^[a-zA-Z0-9_.$#|@/:;=+'-]+$`. */
```

Comment:
If this doesn't validate does the caller get an Exception back?

Reply:
Yep, I tried it in a test and got
```
[info]   'java.lang.Exception: 'document_id' failed validation with the error ''!@#$%^&*()' contains invalid characters'' is a Left, expected a Right. (FullInte
grationTest.scala:253)
```